src/map/script.cpp

```diff
@@ -24180,12 +24181,42 @@ BUILDIN_FUNC(mail){
 			}
 		}
 
+		// Refine
+		if (!script_hasdata(st, 9)) {
+			break;
+		}
+
+		data = script_getdata(st, 9);
+
+		if (!mail_sub(st, data, sd, 9, &name, &start, &end, &id)) {
+			return SCRIPT_CMD_FAILURE;
+		}
+
+		for (i = 0; i < num_items && start < end; i++, start++) {
+			struct item_data* item = itemdb_exists(msg.item[i].nameid);
+
+			ref = (int32)__64BPRTSIZE(get_val2(st, reference_uid(id, start), reference_getref(data)));
```

Comment:
The declaration for `ref` could be moved down inside `if (!script_hasdata(st, 9)) {` or you could just rid of it and directly assign it to `msg.item[i].refine`.